automated-record-card/index.html

```diff
@@ -0,0 +1,196 @@
+            fieldRecordID.removeValue();
+          } else {
+            fieldRecordID.setValue(value);
+          }
+        }
+
+        function unloadHandler () {
+          window.removeEventListener('onbeforeunload', unloadHandler);
+          inputRecordID.removeEventListener('input', keyboardInputHandler);
+          detachValueChangeHandler();
+        }
+
+        // find a value in one of the defined languages
+        function languageValue (value) {
+          const languageKeys = ['en', 'eng', 'def', 'und'];
+          for (let key of languageKeys) {
+            if (value[key]) {
+              return value[key][0];
+            }
+          }
+          return;
+        }
+
+        // clear all fields
+        function clearEntry () {
+          // clear message
+          document.getElementById('message').innerText = '';
+
+          // clear fields
+          extension.entry.fields.name.removeValue();
+          extension.entry.fields.creator.removeValue();
+          extension.entry.fields.provider.removeValue();
+          extension.entry.fields.thumbnailUrl.removeValue();
+
+          document.getElementById('thumbnail-preview').innerHTML = '';
+        }
+
+        // click reset button
+        document.getElementById('reset-record').addEventListener('click', e => {
+          e.preventDefault();
+          clearEntry();
+          // clear identifier
+          extension.entry.fields.identifier.removeValue();
+          valueRecordID = extension.entry.fields.identifier.getValue();
+        });
+
+        // click go button
+        document.getElementById('find-record').addEventListener('click', e => {
+          e.preventDefault();
+
+          // empty fields
+          clearEntry();
+
+          // message
+          document.getElementById('message').innerText = 'Processing';
+
+          // is there a value
+          if (inputRecordID.value === '') {
+            document.getElementById('message').innerText = 'Add a record!';
+            return;
+          }
+
+          const recordId = inputRecordID.value.match('\/[0-9]+\/[a-zA-Z0-9_]+$')[0];
+          const xhr = new XMLHttpRequest();
+
+          xhr.open('GET', 'https://www.europeana.eu/api/v2/record' + recordId + '.json?wskey=[API_KEY]');
+          xhr.send(null);
+          xhr.responseType = 'json';
+
+          xhr.onreadystatechange = function () {
+            if (xhr.readyState === 4) { // 4 = done
+              if (xhr.status === 200) {
+
+                const response = xhr.response.object;
+                let objectTitle;
+
+                // set field values
+                extension.entry.fields.identifier.setValue(recordId);
+                valueRecordID = recordId;
+
+                if (response.proxies) {
+                  if (response.proxies[0].dcTitle) {
+                    objectTitle = languageValue(response.proxies[0].dcTitle);
+                    extension.entry.fields.name.setValue(objectTitle);
```

Comment:
This is also quite brittle. I suspect that we will eventually want a way to be able to specify a map EDM of fields to other form fields on the content type, but I can't think of an obvious way to do so. Perhaps add a TODO note in the meantime.